color_refinement.py

```diff
@@ -4,12 +4,28 @@
 """
 from color_refiment_helper import *
 import time
+import preprocessing
 from graph_io import *
 
 PATH = './graphs/treepaths/'
 GRAPH = 'threepaths160.gr'
 
 
+def preprocess_graphs(g: "Graph", h: "Graph") -> bool:
+    """
+    Pre-processing:
+        - Remove loners
+        - Check for equal vertices
+
+    :param g: first graph to compare
+    :param h: second graph to compare
+    :return: True if graph is eligible for color_refinement and isomorphism(s) detection
+    """
+    preprocessing.remove_loners(g)
+    preprocessing.remove_loners(h)
```

Comment:
Now g and h are changed, which alters the graph on which you eventually do the color refinement. If you remove 3 _loners_ on each graph. The extra iso/auto-morphisms it gives for the loners matching to each other misses..
Possible fix: do you preproccessing with a copy of g and h.
Other possible fix: remove and place back (possibly more work and not needed?)